rmgpy/tools/thermo_group_validation.py

```diff
@@ -0,0 +1,237 @@
+    else:
+        raise Exception("Variable {0} is not a supported thermo attribute for comparison".format(variable))
+
+def execute(path):
+
+    #key is RMG-generated SMILES, value is list with first entry as LibraryEntry
+    #and the rest ThermoData from group estimation
+
+    loadInput(path, speciesThermo)
+    variable = inputDict['variable']
+    heading = ''
+    units =''
+    if variable == 'H':
+        heading = 'Enthalpy'
+        units = 'kcal/mol'
+    elif variable == 'S':
+        heading = 'Entropy'
+        units = 'cal/mol'
+    elif variable == 'Cp':
+        heading = 'Heat_Capacity'
+        units = 'cal/mol'
+    elif variable == 'G':
+        heading = 'Free_Energy'
+        units = 'kcal/mol'
+    T0 = inputDict['temperature']
+    #load database with no libraries
+
+    #change to new directory for RMG-database
+    with cd(settings['database.directory']):
+        for commitStr in inputDict['commitStrs']:
+            with previousCommit(commitStr, settings['database.directory']):
+                newGroups= ThermoDatabase()
+                newGroups.load(os.path.join(settings['database.directory'], 'thermo'))
+                for smiles in speciesThermo:
+                    newSpecies = Species().fromSMILES(smiles)
+                    speciesThermo[smiles].append(newGroups.getThermoDataFromGroups(newSpecies))
+
+    parityData={}
+
+    #get initial dictionary value: [data from library, data from groups of initial commit]
+    for smiles, thermoList in speciesThermo.iteritems():
+        parityData[smiles] = []
+        for thermoData in thermoList:
+            if isinstance(thermoData, LibraryEntry):
+                parityData[smiles].append(parityFunction(variable, thermoData.thermoData, T0))
+            else: parityData[smiles].append(parityFunction(variable, thermoData, T0))
+
+    #make list of highst stdDev of groups (first element is name, second is library value)
+    csvList = [[name]+data for name, data in parityData.iteritems()]
+
+    #Put into kcal/cal
+    conversionCsvList=[]
+    for point in csvList:
+        newPoint=[point[0]]
+        for value in point[1:]:
+            if variable == 'H' or variable == 'G':
+                newPoint.append(value/1000.0/4.182)
+            else:
+                newPoint.append(value/4.182)
+        conversionCsvList.append(newPoint)
+
+    csvList= conversionCsvList
+
+    #Sort based on largest difference in group estimation
+    csvList = sorted(csvList, key = lambda x: std(x[2:]))
+    csvList.reverse()
+
+    with open(os.path.join(os.getcwd(), heading+'.csv'), 'wb') as output:
+        spamwriter=csv.writer(output)
+        #add Header
+        spamwriter.writerow(['SMILES', 'Library Value at '+str(T0)+'K']+ inputDict['commitNames'])
+        for line in csvList:
+            spamwriter.writerow(line)
+
+    #Remove points where group estimates are the same for plotting
+    prunedCsvList=[]
+    #don't include header in pruned csv
```

Comment:
would be nicer if can add more detailed description for data structure of `prunedCsvList ` and `csvList` since they're non-homogeneous list (containing string, float, etc). E.G., comments like each item in those lists is a list, in that list first item is name, second item is library value, third item is the estimated value at commit1, ...